src/view/renderer.js

```diff
@@ -180,23 +187,43 @@ export default class Renderer {
 	render() {
 		let inlineFillerPosition;
 
-		// There was inline filler rendered in the DOM but it's not
-		// at the selection position any more, so we can remove it
-		// (cause even if it's needed, it must be placed in another location).
-		if ( this._inlineFiller && !this._isSelectionInInlineFiller() ) {
-			this._removeInlineFiller();
-		}
-
-		// If we've got the filler, let's try to guess its position in the view.
 		if ( this._inlineFiller ) {
-			inlineFillerPosition = this._getInlineFillerPosition();
+			if ( !this.isComposing && !this._isSelectionInInlineFiller( false ) ) {
+				// There was inline filler rendered in the DOM but it's not at the selection position any more, so we can remove
+				// it if not during composition (cause even if it's needed, it must be placed in another location).
+				this._removeInlineFiller();
+			} else if ( this.isComposing ) {
+				// When selection has 0 ranges, `isCollapsed` returns false, but here
+				// we are only interested in non-collapsed selection (so with at least 1 range).
+				if ( !this.selection.isCollapsed && this.selection.rangeCount > 0 ) {
+					// During composition selection may be extended and its start/end moved to a different text
+					// node (using 'shift + up' in Chrome on Windows during composition). In such situations
+					// filler should not be moved or deleted (because it is possible to continue composing).
+					inlineFillerPosition = this._getExistingInlineFillerPosition();
+				} else if ( !this._isValidCompositionSelection() ) {
```

Comment:
This function is used only in this place so it could be made more contextual. Like `_checkCompositionWasNotEndedCorrectly()`.